simapp/export.go

```diff
@@ -191,16 +192,24 @@ func (app *SimApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs []
 	}
 
 	// iterate through unbonding delegations, reset creation height
-	err = app.StakingKeeper.IterateUnbondingDelegations(ctx, func(_ int64, ubd stakingtypes.UnbondingDelegation) (stop bool) {
-		for i := range ubd.Entries {
-			ubd.Entries[i].CreationHeight = 0
-		}
-		err = app.StakingKeeper.SetUnbondingDelegation(ctx, ubd)
-		if err != nil {
-			panic(err)
-		}
-		return false
-	})
+	err = app.StakingKeeper.UnbondingDelegations.Walk(
+		ctx,
+		nil,
+		func(key collections.Pair[sdk.AccAddress, sdk.ValAddress], value []byte) (stop bool, err error) {
+			ubd, err := stakingtypes.UnmarshalUBD(app.appCodec, value)
+			if err != nil {
+				return true, err
+			}
```

Comment:
why are we doing this again? can't we set `UnbondingDelegation` directly here instead of `[]byte`

Reply:
we are using []byte value for ubd while setting UnbondingDelegations in keeper

Reply:
Why are we setting `[]btye` value in keeper?

Reply:
changed to UnbondingDelegations